prime-router/metadata/HL7/catchall/hl7/datatypes/XAD/XADExtension.yml

```diff
@@ -30,6 +30,18 @@ extension:
           type: STRING
           valueOf: XAD.2
           expressionType: HL7Spec
+    - expressionType: nested
+      vars:
+        xad6: STRING_ALL, XAD.6
+      condition: $xad6 NOT_NULL
+      expressionsMap:
+        url:
+          type: STRING
+          value: XAD.6
+        valueString:
+          type: STRING
+          valueOf: XAD.6
+          expressionType: HL7Spec
```

Comment:
This should have been a good idea and extra coverage to something that's already mapped similar to OBR.25. However, adding the FHIR->HL7 mappings for this field made a subset of our integration tests go _wild_[1]. They're all from the [translation test config file](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/testIntegration/resources/datatests/translation-test-config.csv). 

When adding XAD.6 to FHIR->HL7 XADExtension mappings, the tests all showed that the country was missing from the final HL7 file. I'm not exactly sure what is going on and I haven't been able to get the formatting the fhirdata command to work with whatever kind of schema is being used to create these files[2]. I thought that if I could generate the expected fhir then it might answer what was going on. 
 
[1] Example: 
![Screenshot 2024-07-30 at 6 23 16 PM](https://github.com/user-attachments/assets/849b476d-0bc8-4057-aad7-b7605211b7a5)


[2] I've used this fhirdata command taken from the `recreate-fhir-in-dir.sh` script on some files before if I just wanted it to run on a single file: `./gradlew primeCLI --args="fhirdata --input-file  src/testIntegration/resources/datatests/FHIR_to_HL7/sample_IN_20240618-0001.hl7 --output-file src/testIntegration/resources/datatests/FHIR_to_HL7/sample_IN_20240618-0001.fhir  --output-format FHIR --receiver-schema prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/IN/IN-receiver-transform.yml"`. This fails with `Unexpected scheme: null`, but honestly I'm not even certain if this schema should be run as receiver-schema or enrichment-schema.

Reply:
Huddled with Brick, no resolution for this yet. It was determined the reasoning behind adding XAD.6 to XADExtension is because with `$country` getting mapped to it, there is the possibility that an original HL7 value is lost during conversions. Probably this could be resolved by updating the FHIR->HL7 mappings with a null check. 

Todo: For this and OBR.25 I want to look into the intermediate FHIR data to see why the extension values are being treated as null and overwriting the proper values.

Todo: Update XAD.6 mappings to play well together and not lose the original XAD.6 value

Reply:
Follow up on the translation tests going wild: The tests that failed were only the FHIR -> HL7 tests. This was because XAD.6 in XADExtension was overwriting XAD.6 From XAD/Address due to the order the mappings were processed in (and this was an issue because the FHIR test data did not have the new extension added to it). 

Resolution from standup yesterday: It was determined that MSH.17 as HD/Organization.Country should be mapped inline instead of using XAD/Address. The main reason being that XAD/Address should only be used by HL7 fields of the XAD datatype. By changing HD/Organization.Country to be inlined (a scenario that is only utilized for MSH.17), there is no longer a need for $country to be added to XAD/Address XAD.6 either. This also means that there is no possibility that XAD.6 gets overwritten and thus no need for XADExtension to preserve XAD.6.

Thus, the following two changes are being reverted:
- Add $country to XAD.6
- Add XAD.6 to XADExtension.

Rervert for XAD.6 to XADExtension: https://github.com/CDCgov/prime-reportstream/pull/15333/commits/9e047f8b27e9b416532689c42b82251e29b451e3
Change HD/Org to inline country, and Revert $country to XAD.6: https://github.com/CDCgov/prime-reportstream/pull/15333/commits/24228ee49f3bf663cacc7f99e3f4599ff9a26825
Misc. test updates as a result: https://github.com/CDCgov/prime-reportstream/pull/15333/commits/0a680d94f7b59e97417bd9320227aa4f9398c8bf